src/core/server/http/lifecycle_handlers.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { OnPostAuthHandler } from './lifecycle/on_post_auth';
+import { OnPreResponseHandler } from './lifecycle/on_pre_response';
+import { HttpConfig } from './http_config';
+import { Env } from '../config';
+import { LifecycleRegistrar } from './http_server';
+
+const VERSION_HEADER = 'kbn-version';
+const XSRF_HEADER = 'kbn-xsrf';
+const KIBANA_NAME_HEADER = 'kbn-name';
```

Comment:
What do you think about adding a file for these constants that can be shared across `public` and `server`?

While I was implementing the System API behavior, I was considering adding a `src/core/utils/constants` file for the `kbn-system-api` and `kbn-version` header names.

Reply:
Yes I though about it and was not sure where I was supposed to put the constants file. should it be `src/core/utils` or `src/core/types`? Constants are not exactly one or the other.

I could also exports these from `src/core/server/http` (and `src/core/server`) and then expose them in `src/core/server/types` to be eventually used by the client-side?

Reply:
Other benefit is that if we need to change the header name in the future, we only have to edit it in a single place and don't risk missing a string that needs to be updated (or a string from a yet-to-be-merged PR not getting updated).

That said, I don't imagine these changing often (if at all) or them being used in many more places (if any).

Kind of a toss up on benefits here. I'm fine leaving as-is in this PR and removing my shared module in #53734.

Reply:
Will keep them as-is for now. Can easily be refactored later on if the benefits appear more significants at some point.